packages/block-editor/src/components/link-control/index.js

```diff
@@ -251,59 +369,121 @@ function LinkControl( {
-					<VisuallyHidden>{ SearchResultsLabel }</VisuallyHidden>
-				) }
-
+				{ searchResultsLabel }
 				<div
 					{ ...suggestionsListProps }
 					className={ resultsListClasses }
 					aria-labelledby={ searchResultsLabelId }
 				>
-					{ suggestions.map( ( suggestion, index ) => (
-						<LinkControlSearchItem
-							key={ `${ suggestion.id }-${ suggestion.type }` }
-							itemProps={ buildSuggestionItemProps(
-								suggestion,
-								index
-							) }
-							suggestion={ suggestion }
-							onClick={ () => {
-								onChange( { ...value, ...suggestion } );
-								stopEditing();
-							} }
-							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes(
-								suggestion.type.toLowerCase()
-							) }
-							searchTerm={ inputValue }
-						/>
-					) ) }
+					{ suggestions.map( ( suggestion, index ) => {
+						if (
+							shouldShowCreateEntity &&
+							CREATE_TYPE === suggestion.type
+						) {
+							return (
+								<LinkControlSearchCreate
+									searchTerm={ inputValue }
+									onClick={ async () => {
+										try {
+											cancelableOnCreate.current = makeCancelable(
+												handleOnCreate(
+													suggestion.title
+												)
+											);
+
+											await cancelableOnCreate.current
+												.promise;
+											// Only stop editing if not rejected
+											stopEditing();
+										} catch ( error ) {
+											if ( error && error.isCanceled ) {
+												return; // bail if canceled to avoid setting state
+											}
+
+											setErrorMessage(
+												__(
+													'An unknown error occurred during creation. Please try again.'
+												)
+											);
+											setIsResolvingLink( false );
+											setIsEditingLink( true );
+										}
+									} }
+									// Intentionally only using `type` here as
+									// the constant is enough to uniquely
+									// identify the single "CREATE" suggestion.
+									key={ suggestion.type }
+									itemProps={ buildSuggestionItemProps(
+										suggestion,
+										index
+									) }
+									isSelected={ index === selectedSuggestion }
+								/>
+							);
+						}
+
+						// If we're not handling "Create" suggestions above then
+						// we don't want them in the main results so exit early
+						if ( CREATE_TYPE === suggestion.type ) {
+							return null;
+						}
+
+						return (
+							<LinkControlSearchItem
+								key={ `${ suggestion.id }-${ suggestion.type }` }
+								itemProps={ buildSuggestionItemProps(
+									suggestion,
+									index
+								) }
+								suggestion={ suggestion }
+								index={ index }
+								onClick={ () => {
+									stopEditing();
+									onChange( { ...value, ...suggestion } );
+								} }
+								isSelected={ index === selectedSuggestion }
+								isURL={ directLinkEntryTypes.includes(
+									suggestion.type.toLowerCase()
+								) }
+								searchTerm={ inputValue }
+							/>
+						);
```

Comment:
It would be nice if we could refactor to reduce the code duplication between this and the `LinkControlSearchCreate` above. It seems like the two receive largely the same props, with few exceptions.

Reply:
Agreed. If no one objects, I think this is one for a followup refactor Issue + PR.

Reply:
Only three props are actually shared. I really think given the potential for rebases we should do this in a followup.

Reply:
Follow up Issue https://github.com/WordPress/gutenberg/issues/20271